RadixWallet/Features/AssetsFeature/Components/StakeUnitList/Components/StakeClaimNFTsView.swift

```diff
@@ -82,24 +65,80 @@ public struct StakeClaimNFTSView: View {
 				Spacer()
 			}
 
-			if !viewState.unstaking.isEmpty {
-				sectionView(viewState.unstaking, kind: .unstaking)
-			}
-
-			if !viewState.readyToBeClaimed.isEmpty {
-				sectionView(viewState.readyToBeClaimed, kind: .readyToBeClaimed)
-			}
-
-			if !viewState.toBeClaimed.isEmpty {
-				sectionView(viewState.toBeClaimed, kind: .toBeClaimed)
-			}
+			StakeClaimTokensView(
+				viewState: viewState.stakeClaimTokens,
+				background: background,
+				onTap: onTap,
+				onClaimAllTapped: onClaimAllTapped
+			)
 		}
 		.padding(.medium3)
 		.background(background)
 	}
+}
+
+// MARK: - StakeClaimTokensView
+public struct StakeClaimTokensView: View {
+	enum SectionKind {
+		case unstaking
+		case readyToBeClaimed
+		case toBeClaimed
+	}
+
+	public struct ViewState: Sendable, Hashable {
+		public let canClaimTokens: Bool
+		public let stakeClaims: IdentifiedArrayOf<OnLedgerEntitiesClient.StakeClaim>
```

Comment:
I presume that the purpose of the IdentifiedArrays here is to ensure uniqueness, but I don't think that is needed in the ViewState. It should be enough to do that in State.